src/Umbraco.Web/PropertyEditors/ValueConverters/MemberPickerPropertyConverter.cs

```diff
@@ -32,36 +32,32 @@ public override object ConvertDataToSource(PublishedPropertyType propertyType, o
             var attemptConvertInt = source.TryConvertTo<int>();
             if (attemptConvertInt.Success)
                 return attemptConvertInt.Result;
+
             var attemptConvertUdi = source.TryConvertTo<Udi>();
             if (attemptConvertUdi.Success)
                 return attemptConvertUdi.Result;
+
             return null;
         }
 
         public override object ConvertSourceToObject(PublishedPropertyType propertyType, object source, bool preview)
         {
-            if (source == null)
-                return null;
+            var umbracoContext = UmbracoContext.Current;
+            if (umbracoContext == null)
+                return source;
 
-            if (UmbracoContext.Current == null) return source;
+            var membershipHelper = new MembershipHelper(umbracoContext);
 
-            var umbracoHelper = new UmbracoHelper(UmbracoContext.Current);
-            IPublishedContent member;
-            if (source is int)
-            {                    
-                member = umbracoHelper.TypedMember((int)source);
-                if (member != null)
-                    return member;
+            if (source is int sourceInt)
+            {
+                return membershipHelper.GetById(sourceInt);
             }
-            else
+            else if (source is GuidUdi sourceUdi)
             {
-                var sourceUdi = source as Udi;
-                member = umbracoHelper.TypedMember(sourceUdi);
-                if (member != null)
-                    return member;
+                return membershipHelper.GetByProviderKey(sourceUdi.Guid);
```

Comment:
Should this also check for the entity type (as done for content and media)? `UmbracoHelper.TypedContent(Udi)` doesn't contain this check, so that's why I've omitted it here.